src/app/code/community/Meanbee/ConfigPoweredCss/controllers/Adminhtml/ConfigPoweredCssController.php

```diff
@@ -7,20 +7,13 @@ public function regenerateAction()
         /** @var Mage_Adminhtml_Model_Session $session */
         $session = Mage::getSingleton('adminhtml/session');
 
-        /** @var Meanbee_ConfigPoweredCss_Model_Css $css */
-        $css = Mage::getModel('meanbee_configpoweredcss/css');
-
-        $stores = Mage::app()->getStores();
-        foreach ($stores as $storeId => $store) {
-            try {
-                Mage::app()->setCurrentStore($storeId);
-                $css->publish($storeId);
-            } catch (Exception $e) {
-                Mage::app()->setCurrentStore(Mage_Core_Model_App::ADMIN_STORE_ID);
-                $session->addError($e->getMessage());
-                $this->_redirect('*/cache');
-                return;
-            }
+        try {
+            Mage::helper('meanbee_configpoweredcss')->regenerateConfigPoweredCss();
+        } catch (Exception $e) {
+            Mage::app()->setCurrentStore(Mage_Core_Model_App::ADMIN_STORE_ID);
```

Comment:
I think this line belongs in the helper method, because of the division of responsibility. The helper method sets a different store during generations, so it should have the responsibility of resetting it afterwards. The controller shouldn't need the knowledge that the store could have/has been changed and needs to be reset.

Moreover, since the helper method can be used in any context, it should record the current store at the start and reset to it, instead of always resetting to the admin store. That's just protection against possible bugs in the future.

Reply:
Fair point, I partially agree. I'll remove the later call as we shouldn't have to set the current store if  we've not had an exception, that can be done by the helper, however, in the case that we end up with an exception, it would be good that the controller sets the store to admin.

Reply:
I don't see how that improves things though. The controller shouldn't be responsible for resetting the environment in any case. The helper method shouldn't screw up the environment for the user, and leave side effects, whatever the user is. It's perfectly fine to catch the exception in the helper method, reset the environment, and then throw the exception forward again with `throw $e;`. In fact, the clearest change would be to use the `finally` keyword in the helper method, without catching the exception, i.e.:

```
try {
    // Do the work that messes with the environment
} finally {
    // Reset the environment
}
```